### PR TITLE
add push button

### PR DESCRIPTION
**Pull request by ckline-tryptic**

### CODE REVIEW

File: client/src/test/suite/extension.test.ts [hunk 1]
```diff
@@ -306,4 +313,34 @@ suite('Extension Test Suite', function (this: Suite) {
...
   })
+  test('push with no conflict', async () => {
+    stubRepo.commit = sinon.stub().throws()
+    stubRepo.pull = sinon.stub().throws()
+    stubRepo.push = sinon.stub().throws()
+    await pushContent.pushContent()
+  })
+  test('push with merge conflict', async () => {
+    try {
+      stubRepo.commit = sinon.stub().throws(GitErrorCodes.Conflict)
```
Comment (tomjw64) [inline client/src/test/suite/extension.test.ts:324]: the `commit` function can never throw an error formed this way. Also, `pushContent` should **not** throw in the case of a conflict, so the assertion in this test doesn't seem correct

File: client/src/test/suite/extension.test.ts [hunk 2]
```diff
@@ -306,4 +313,34 @@ suite('Extension Test Suite', function (this: Suite) {
...
+      stubRepo.commit = sinon.stub().throws(() => {
+        const error: any = new Error()
+        error.stdout = 'nothing to commit '
+        return error
+      })
+      stubRepo.pull = sinon.stub().throws()
+      stubRepo.push = sinon.stub().throws()
+      await pushContent.pushContent()
+    } catch (e) {
+      assert((e.stdout as string).includes('nothing to commit'))
```
Comment (tomjw64) [inline client/src/test/suite/extension.test.ts:343]: `pushContent` also should not throw in this case

**Review by tomjw64 — CHANGES_REQUESTED** [review]

**Comment by ckline-tryptic** [comment]
Wrote tests, but code Codecov was complaining, and code that it looks like my tests would cover are reported as not covered. Making some changes to resolve this.

File: client/src/push-content.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+export const getRepo = (): Repository => {
+  const gitExtension = expect(vscode.extensions.getExtension<GitExtension>('vscode.git')).exports
+  const api = gitExtension.getAPI(1)
+  const result: Repository = api.repositories[0]
+  return result
+}
+
+export const pushContent = () => async () => {
+  const repo = getRepo()
```
[the remark this answers is not in the document]
Comment (ckline-tryptic) [reply client/src/push-content.ts:13]: Mayhaps, it's mainly the repo itself that needs mocking.

File: LICENSE2 [hunk 4]
```diff
@@ -0,0 +1,21 @@
+MIT License
```
[the remark this answers is not in the document]
Comment (ckline-tryptic) [reply LICENSE2:1]: Moving the interface and license files into their own directory.

**Pull request closed without merging**